standards/SPIFFE-ID.md

```diff
@@ -135,7 +141,47 @@ In order for an SVID to be considered valid, it MUST leverage a document type fo
 
 Please see the [X.509 SPIFFE Verifiable Identity Document](X509-SVID.md) or the [JWT SPIFFE Verifiable Identity Document](JWT-SVID.md) specification for more information.
 
-## 4. Conclusion
-The specifications contained within this document cover what it means to be SPIFFE compliant. While other specifications will need to be referenced in order to build a complete implementation, conformance to this document is sufficient for compliance purposes.
+## 4. Security Considerations
+This section includes security considerations that implementers and users should take into account when using SPIFFE IDs and SVIDs.
+
+### 4.1. SVID Assertions
+SVIDs always carry within them a set of data - at the minimum, a SPIFFE ID. Sometimes, this data represents assertions made by the trust domain authority about the SVID's subject. When interpreting meaning from this data, care must be taken to ensure that all parties involved well understand the meaning and significance of the utilized information.
+
+There are four major concerns when considering the relative safety of any given assertion. First is the temporal accuracy - SVIDs are good for some time before they expire, is the assertion in the SVID true over the entirety of the SVID lifetime? Second, there is the scope and influence of the assertion - under what context was the assertion originally made, and how far reaching should its influence be? Third is the issue of interpretation and meaning - does the assertion have the same meaning to the authority and consumer, or is there room for divergent interpretation? And finally, the veracity of the assertion itself may be called into question in some extreme cases.
+
+This section explores all four areas of concern, and provides guidelines by which operators may assess the relative safety of any given SVID assertion. Generally speaking, operators should err on the side of caution and only include assertions in which there is a very high degree of confidence in the safety of the assertion in question. Assertions that are directly formalized by the SPIFFE specification are considered safe as they have been closely scrutinized.
```

Comment:
In what cases would we not trust another domain's issued SPIFFE IDs (for instance) but still say that we trust the domain at all? I guess maybe what the intention here is is to say there is a difference between "core" attributes which that come with "I trust this domain" vs extra attributes that come with considerations and allowlists/blocklists.

Reply:
I think certain trust assumptions come with trusting a domain, the core one being ID. If you do not trust the IDs coming out of a domain, then I don't think you can trust anything coming from the domain.

Reply:
> I think certain trust assumptions come with trusting a domain, the core one being ID. If you do not trust the IDs coming out of a domain, then I don't think you can trust anything coming from the domain.

I agree with this statement, but I think it is worth stating explicitly, rather than the current language.

Reply:
I feel like the SPIFFE ID is a bit of a special case. In many ways, its value is completely arbitrary and does not matter when considering the loosely coupled federation use cases.

For example, when federating with a vendor that you send data to for indexing, the business has already made a decision to trust this vendor (the entirety of this vendor - the legal entity) with its data. Nobody in this business is saying they trust this data to a specific workload running within the vendor, instead the domain of trust is the vendor as a whole. But, we still probably want to validate the vendor's SPIFFE ID (beyond trust domain) right?

I think the reality is that this validation is a quick and easy win that will probably make a measurable improvement .. however if the SPIFFE ID is lied about (by the vendor, to be the "correct" one), it doesn't really matter because the net effect is the same, and the trust expectations of the business have not been violated.

So I guess the argument I'm trying to make here is that, maybe "trusting the SPIFFE ID" is a moot point because even if it is not trustworthy, lying about it or otherwise does not actually violate any of the models I can think of? It is effectively equivalent to a trust domain compromise at that point.

I have more follow-on thoughts that will bring me back to the original statement questioned in this comment, but I think I'll stop short of there because I'd like to see what people think of the above first

Reply:
> So I guess the argument I'm trying to make here is that, maybe "trusting the SPIFFE ID" is a moot point because even if it is not trustworthy, lying about it or otherwise does not actually violate any of the models I can think of? It is effectively equivalent to a trust domain compromise at that point.

Thinking from the perspective here of "this other trust domain is an entirely different business I have no view into" for me adds clarity to the wording. We have a business agreement with `B`, and a technical implementation agreement with `B` that `spiffe://b/workload/foo` will get access to our internal resource `bar`. How `B` engineers decide internally what qualifies as `spiffe://b/workload/foo` in standard operations, experimentation, mitigation scenarios, total hacks, or migrations is entirely up to them and we have no view of whether the `B` engineers are "lying" or not because we don't know their operations to begin with. Nor do we care about the gritty internal operations because our agreement is simply "we will allow `spiffe://b/workload/foo`, and you will follow as best of practices as you can to ensure integrity of issuance of `spiffe://b/workload/foo`".

Reply:
I mean, either you care which SPIFFE-ID is on the other end of the line, or you don't.  If you don't care, it doesn't make any sense to validate it, and you might still decide to federate with the domain if all you want is a sort of organization-level mutual-auth. (Or, one side might care while the other does not.)

However, I don't think we should assume that is it generally the case that you don't care about the SPIFFE-ID for federation use cases.  If SPIFFE is widely deployed in an organization then there could be many different applications in the same trust domain.  

To make a real world analogy, let's say you're working with an accounting firm, and you give highly confidential revenue and earning statements to your accountant, Alicia.  One day when you walk in to drop them off, Steve from the cafeteria meets you in the lobby and says Susan is out today, but you should leave the documents with him instead.  Steve has an authentic ID badge issued by the company.  Maybe you've even seen him in the cafeteria slicing bagels, so you have no doubt he does actually work for the accounting firm.  Do you leave the highly sensitive documents with him?  Does the fact that Steve has done this translate into "the accounting firm is compromised" in your mind?

I think we may have in our mind very near-term use cases where SPIFFE control planes are managed by a single application team, or possibly an infrastructure team that serves a small group of related applications.  But thinking beyond that, it might be desirable for there to be a single, centrally managed SPIFFE control plane for a large multinational.  Maybe it even has self-service to allow your little prototype application to get a SPIFFE-ID in some subspace of the  domain SPIFFE-ID space.

Reply:
We had a discussion on this today in the SIG-Spec call:

- Everyone agrees that we should not be telling our users to blindly trust the veracity of any given SPIFFE ID "assertion"
- We discussed the fact that trustworthiness of the SPIFFE ID "assertion" is transcended by the trustworthiness of the entity with which you are federating
- Everyone agrees that just because you consider the entity you are federating with to be trustworthy, you should still validate the SPIFFE ID
- If you do not find the entity to be trustworthy, and therefor do not find the SPIFFE ID to be trustworthy, then you should probably not be sending your data to this entity to begin with

Which is aligned with previous statements:
>> I think certain trust assumptions come with trusting a domain, the core one being ID. If you do not trust the IDs coming out of a domain, then I don't think you can trust anything coming from the domain.
>
> I agree with this statement, but I think it is worth stating explicitly, rather than the current language.

Though the above can also be adjusted from "trust anything coming from the domain" to "trust sending any data into the domain".

I _think_ we're all actually aligned here, and everyone on the call was amenable to language similar to that in the last bullet point above. That said, after reviewing this paragraph and surrounding text, I think my suggestion is actually to just strike the offending sentence. If one holds even the SPIFFE specified assertions to the same guidelines we've outlined for everything else here, the same conclusions we're discussing now should be reached.

Alternatively, if we want to call this out directly, I think we should have a new paragraph:
```suggestion
This section explores all four areas of concern, and provides guidelines by which operators may assess the relative safety of any given SVID assertion. Generally speaking, operators should err on the side of caution and only include assertions in which there is a very high degree of confidence in the safety of the assertion in question.

It should be noted that while assertions which are directly formalized by the SPIFFE specifications are not typically vulnerable to problems related to interpretation and meaning, they can still be vulnerable to problems related to veracity. However, due to the tightly scoped nature of SPIFFE defined assertions, veracity concerns in this regard indicate much larger concerns around the security posture of the trust domain in question, at which point operators should seriuosly question whether or not data should be exchanged with these systems in the first place.
```

Reply:
> sending any data into the domain

small nit: I think trust extends to both sending data into and receiving data from a federated trust domain.

Reply:
I pushed an update that includes the alternate language, which I think Spike may prefer since it directs readers to call federation relationships into question if they feel that they can't trust SPIFFE IDs in a foreign trust domain